### PR TITLE
Changes for TanninOne/modorganizer#309 (3 of 3)

### PR DESCRIPTION
**Pull request by ThosRTanner**
Changes for TanninOne/modorganizer#309

This does some cleanup of the spawn code so it behaves the same when spawning from the main code or a plugin, and adds a little extra detail to the 'remember me' question box

Also some unrelated build cleanups for scons and include-what-you-use

### CODE REVIEW

**Comment by ThosRTanner** [comment]
A note:

If you want to individually configure each executable for running/not running steam you'll have to clear the remembered dialogue buttons or hand edit your modorganizer.ini file

File: src/organizercore.cpp [hunk 1]
```diff
@@ -892,66 +909,22 @@ QStringList OrganizerCore::modsSortedByProfilePriority() const
...
+    m_UserInterface->lock();
+  }
+  ON_BLOCK_EXIT([&] () {
+    if (m_UserInterface != nullptr) { m_UserInterface->unlock(); }
+  });
 
   HANDLE processHandle = spawnBinaryDirect(binary, arguments, m_CurrentProfile->name(), currentDirectory, steamAppID);
   if (processHandle != INVALID_HANDLE_VALUE) {
-    if (closeAfterStart && (m_UserInterface != nullptr)) {
+    if (closeAfterStart && m_UserInterface != nullptr) {
```
Comment (TanninOne) [inline src/organizercore.cpp]: Please respect the existing coding style and please don't make unnecessary changes in code that I will have to merge into the branch.
In general, it would have been easier if this diff had been made to the new_vfs_library branch because there has been some changes to the process spawning code.

File: src/organizercore.cpp [hunk 2]
```diff
@@ -1599,43 +1592,3 @@ void OrganizerCore::prepareStart() {
   storeSettings();
 }
 
-/*
```
Comment (TanninOne) [inline src/organizercore.cpp]: This is an unnecessary diff. This code is actually used (in modified form) in the branch so this diff will cause an avoidable conflict.
